File: typescript/src/models/ingestEventsResponse.ts
// this file is @generated
import { type IngestFailure, IngestFailureSerializer } from "./ingestFailure";

export interface IngestEventsResponse {
  /** Events that failed to ingest. Omitted when no failures. */
  failures?: IngestFailure[];
}

export const IngestEventsResponseSerializer = {
  _fromJsonObject(object: any): IngestEventsResponse {
    return {
      failures:
        object["failures"] != null
          ? object["failures"].map((item: any) =>
              IngestFailureSerializer._fromJsonObject(item)
            )
          : undefined,
    };
  },

  _toJsonObject(self: IngestEventsResponse): any {
    return {
      failures:
        self.failures != null
          ? self.failures.map((item: any) => IngestFailureSerializer._toJsonObject(item))
          : undefined,
    };
  },
};

File: python/meteroid/models/ingest_events_response.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .ingest_failure import IngestFailure


@dataclasses.dataclass
class IngestEventsResponse(BaseModel):
    failures: t.Optional[t.List[IngestFailure]] = None
    """Events that failed to ingest. Omitted when no failures."""
